LLM05-LLM10 ITA (#69)
talesh/www-project-top-10-for-large-language-model-applications: 1_1_vulns/it/LLM10_ModelTheft.md

Il furto di un LLM (Modelli di Linguaggio di Grandi Dimensioni) rappresenta una preoccupante questione di sicurezza, divenendo i modelli linguistici sempre più potenti e diffusi. Organizzazioni e ricercatori, devono dare priorità a misure di protezione adeguate volte a proteggere i loro modelli LLM, garantendo la riservatezza e l'integrità della loro proprietà intellettuale. È cruciale adottare un sistema di sicurezza completo (es. sistema di gestione della sicurezza delle informazioni o altre pratiche equivalenti) che includa controlli di accesso, crittografia e monitoraggio continuo, per mitigare i rischi associati al furto del modello LLM, e salvaguardare gli interessi sia degli individui che delle organizzazioni, che ad esso si affidano.

### Esempi Comuni di Vulnerabilità

1. Un aggressore sfrutta una vulnerabilità nell'infrastruttura di un'azienda per accedere senza autorizzazione al loro archivio (repository) di modelli LLM grazie a configurazioni errate nella sicurezza della rete o delle applicazioni.
2. Uno scenario di minaccia interna in cui, un dipendente insoddisfatto, divulga modelli o artefatti correlati ai modelli.

> 3. Un attaccante interroga l'API del modello, utilizzando input meticolosamente elaborati e tecniche di iniezione di prompt (comandi e parametri da linea di comando), per raccogliere un numero sufficiente di output (risposte del modello) utilizzandoli per creare successivamente un modello ombra (shadow model).

Author: A simple recall of the command line nature of prompt injection, being it via API or frontend. Similarly to "LLM" acronym, was just a way to explain the concept. (could be superfluous)

   1. Questo è particolarmente vero per i primi tre casi che più comunemente potrebbero causare questa vulnerabilità quali minacce interne (insider threat), configurazione errata e/o controlli di sicurezza deboli, relativi all'infrastruttura che ospita modelli, pesi e architettura dell'LLM, in cui un attore malintenzionato potrebbe infiltrarsi dall'interno, o dall'esterno.
   2. La gestione dei fornitori, il controllo, la verifica e il controllo del livello di dipendenza da essi sono elementi importanti di attenzione per prevenire gli abusi ed attacchi diretti sulla catena di fornitura.
2. Limitare l'accesso dell'LLM alle risorse di rete, ai servizi interni e alle API.
   1. Questa pratica è efficace per tutti gli esempi di attacchi comuni, fornendo copertura dai rischi e dalle minacce interne (insider threats), e limitando allo stesso tempo ciò a cui l'applicazione LLM "_ha accesso_", rappresentando dunque un meccanismo o metodo preventivo, per prevenire o interrompere attacchi laterali (side-channel attacks).

> 3. Utilizzare un Inventario o Registro centralizzato per i modelli di Machine Learning (ML Models) utilizzati in produzione. Avere un registro di modelli centralizzato impedisce l'accesso non autorizzato ai modelli di ML tramite controlli di accesso, autenticazione e grazie alla capacità di monitoraggio/registrazione, elemnti fondamentali per il governo di questi processi (governance). Avere un repository centralizzato è inoltre vantaggioso per collezionare dati sugli algoritmi utilizzati dai modelli, ai fini della conformità, valutazione dei rischi e loro mitigazione.

Reviewer: ```suggestion
3. Utilizzare un Inventario o Registro centralizzato per i modelli di Machine Learning utilizzati in produzione. Avere un registro di modelli centralizzato impedisce l'accesso non autorizzato ai modelli di ML tramite controlli di accesso, autenticazione e grazie alla capacità di monitoraggio/registrazione, elementi fondamentali per la governance di questi processi. Avere un repository centralizzato è inoltre vantaggioso per raccogliere dati sugli algoritmi utilizzati dai modelli ai fini di conformità, valutazione dei rischi e loro mitigazione.
```

Author: I tried to follow the rule of "not using anglophone terms without at least a translation next to them".

4. Monitorare e verificare regolarmente i registri (log) di accesso e le attività relative agli archivi dei modelli LLM, per rilevare e rispondere tempestivamente a qualsiasi comportamento sospetto o non autorizzato.

> 5. Automatizzare la distribuzion delle cosiddette MLOps (Gestione Operativa dei Modelli di Machine Learning) attraverso flussi di lavoro per la governance, il tracciamento e l'approvazione, per rafforzare i controlli di accesso e rilascio, all'interno dell'infrastruttura.

Reviewer: ```suggestion
5. Automatizzare la distribuzion delle cosiddette MLOps (Gestione Operativa dei Modelli di Machine Learning) attraverso flussi di lavoro per la governance, il tracciamento e l'approvazione, per rafforzare i controlli di accesso e rilascio all'interno dell'infrastruttura.
```

6. Implementare controlli e strategie di mitigazione per ridurre e|o contenere il rischio di attacchi indiretti (side channel) causati da tecniche di iniezione di prompt (prompt injection).
7. Limitare la frequenza delle chiamate API dove applicabile e/o utilizzare filtri per ridurre il rischio di esfiltrazione dati dalle applicazioni LLM, o implementare tecniche (ad es. sistemi di Data Loss Prevention) per rilevare attività di estrazione anche da altre fonti.

> 8. Implementare un addestramento avversariale mirato alla resilienza, sia per aiutare a rilevare query di estrazione, e rafforzare le misure di sicurezza fisica.

Reviewer: ```suggestion
8. Implementare un addestramento avversariale mirato alla resilienza per aiutare a rilevare query di estrazione e rafforzare le misure di sicurezza fisica.
```

> 9. Implementare un framework di marcatura modale e|o temporale (watermarking) nelle fasi di integrazione (input) e monitoraggio (output, trasformazione), parte del ciclo di vita di un LLM.

Reviewer: ```suggestion
9. Implementare un framework di marcatura modale e/o temporale (watermarking) nelle fasi di integrazione (input) e monitoraggio (output, trasformazione), parte del ciclo di vita di un LLM.
```

### Esempi di Scenario di Attacco

1. Un attaccante abusa una vulnerabilità nell'infrastruttura di un'azienda per accedere senza autorizzazione al loro repository di modelli LLM. L'attaccante procede quindi all'esfiltrazione di modelli LLM di valore e li utilizza per lanciare un servizio concorrente di elaborazione del linguaggio o estrarre forzatamente informazioni sensibili, causando gravi danni finanziari all'azienda proprietaria del modello.

> 2. Un dipendente insoddisfatto divulga modelli o artefatti correlati. L'esposizione pubblica che questo scenario rappresenta, aumenta la conoscenza di attaccanti ulteriori, consentendo attacchi avversariali di tipo "gray box" (conoscenza parziale) o, in alternativa, consente di rubare direttamente la proprietà intellettuale esposta.

Reviewer: ```suggestion
2. Un dipendente insoddisfatto divulga modelli o artefatti correlati. L'esposizione pubblica che questo scenario rappresenta aumenta la conoscenza a disposizione di possibili attaccanti, consentendo attacchi avversariali di tipo "gray box" (conoscenza parziale) o, in alternativa, consente di rubare direttamente la proprietà intellettuale esposta.
```

3. Un attaccante interroga l'API con input accuratamente selezionati e raccoglie un numero sufficiente di output per creare un modello ombra (shadow).

> 4. Una mancanza, nei controlli di sicurezza della catena di fornitura (supply chain) porta a perdite di dati relativi a informazioni proprietarie del|sul modello.

Reviewer: ```suggestion
4. Una mancanza nei controlli di sicurezza della catena di fornitura (supply chain) porta a perdite di dati relativi a informazioni proprietarie del o sul modello.
```

> 5. Un attaccante malintenzionato elude le tecniche di filtraggio degli input e negli stadi iniziali dell'LLM, per eseguire un attacco indiretto|laterale e recuperare informazioni sul modello verso una risorsa remota, sotto il suo controllo.

Reviewer: Not 100% sure what "preambles of the LLM" means here, but I would interpret it as the initial instructions given to the LLM (e.g. the part of the prompt preceding the user-provided text). Would that make sense?

```suggestion
5. Un attaccante malintenzionato elude le tecniche di filtraggio degli input e le istruzioni iniziali dell'LLM, per eseguire un attacco indiretto/laterale e recuperare informazioni sul modello caricandole su una risorsa remota sotto il suo controllo.
```
